common/src/main/java/com/wynntils/screens/maps/MainMapScreen.java

```diff
@@ -355,18 +370,22 @@ public boolean doMouseClicked(double mouseX, double mouseY, int button) {
                 if (hovered.hasStaticLocation()) {
                     if (hovered instanceof IconPoi iconPoi) {
                         if (iconPoi instanceof CustomPoi customPoi) {
-                            Models.Compass.setCompassLocation(
-                                    new Location(hovered.getLocation()), iconPoi.getIcon(), customPoi.getColor());
+                            Models.Marker.USER_WAYPOINTS_PROVIDER.addLocation(
```

Comment:
Are we okay with not clearing the old waypoint in these cases? It is a pretty steep change compared to the old behavior; that means you will have to actively clear the old waypoint, or you'll end up with multiple ones.

I am not sure this is okay. At this point we should probably keep the old behavior and clear all old markers before adding a new. Later on, we can rework the UI so you can select if you want to add a new waypoint, or replace the existing.

Reply:
If you check this class now, and the i18n, I added support for the new behavior, and restored the old one.

Reply:
I think we will need to add something along the lines of "if you press shift while you click, you create an additional marker, otherwise you'll replace the old".

The problem with that is that I think we have already "used up" many such combos for other purposes, so there are not many left. But maybe like Alt + click for adding instead of replacing?

Reply:
That was why I recommended that we keep the old "replace" behavior for now and then we can add a way to set additional markers as a follow up. But if you think you can find some free combo, that is at least somewhat reasonable, go ahead.

Reply:
Shift + old keys should be fine.